File: LearnBinder/app/src/main/java/cn/yui/learnbinder/MainActivity.java
package cn.yui.learnbinder;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

import cn.yui.ipclib.core.BinderIPC;
import cn.yui.learnbinder.data.UserInfo;
import cn.yui.learnbinder.ipc.UserInfoSingleton;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        BinderIPC.getInstance().open(this);
        BinderIPC.getInstance().register(UserInfoSingleton.class);
    }

    public void gotoSecondActivity(View view) {
        UserInfoSingleton.getInstance().setUserInfo(new UserInfo("password", "name"));
        startActivity(new Intent(this, SecondActivity.class));
    }
}

File: Demo/app/src/main/java/cn/yui/demo/MainActivity.kt
package cn.yui.demo

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import dagger.hilt.android.AndroidEntryPoint

/**
 * @Description: 主页面
 * @Author: Yui Master
 * @CreateDate: 2022/1/15 7:23 下午
 */
@AndroidEntryPoint
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }
}